### PR TITLE
Core functional implementation

### PR DESCRIPTION
**Pull request by maruraru**
I'm not sure about modules creation in the lib directory, but I definitely need to somehow separate the code from the controller. If I did wrong I'd be grateful for the advice on where to put this new code

### CODE REVIEW

**Comment by AlexeyBoldak** [comment]
You can merge 2 files in lib dir, because they do only 1 thing, so no need to split it. Also If they require some params you should require these params. Also, you need to handle exceptions. And test coverage for new logic needed.

**Comment by maruraru** [comment]
> You can merge 2 files in lib dir, because they do only 1 thing, so no need to split it. Also If they require some params you should require these params. Also, you need to handle exceptions. And test coverage for new logic needed.

done

File: app/controllers/main_page_controller.rb [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 class MainPageController < ActionController::Base
 
+  include RouteBuilding
+
   def index; end
 
+  def create
+    params.require(%i[length_matrix address_array])
```
Comment (AlexeyBoldak) [inline app/controllers/main_page_controller.rb]:
```
def create
  params.require(%i[length_matrix address_array])
    addresses = JSON.parse params[:address_array]
    @result = make_address_sequence(build_route(parse_length_matrix(params)), addresses)
    puts @result

  rescue ActionController::ParameterMissing
    render plain: 'Адрес не введен'
end
```

File: app/controllers/main_page_controller.rb [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+  rescue ActionController::ParameterMissing
+    render plain: 'Адрес не введен'
+  else
+    addresses = JSON.parse params[:address_array]
+    @result = make_address_sequence(build_route(parse_length_matrix(params)), addresses)
+    puts @result
```
Comment (AlexeyBoldak) [inline app/controllers/main_page_controller.rb]: Why puts?

File: config/database.yml [hunk 3]
```diff
@@ -1,25 +1,86 @@
...
+  database: myapp_development
+
+  # The specified database role being used to connect to postgres.
+  # To create additional roles in postgres see `$ createuser --help`.
+  # When left blank, postgres will use the default role. This is
+  # the same name as the operating system user that initialized the database.
+  username: maya
+
+  # The password associated with the postgres role (username).
+  password: root
```
Comment (AlexeyBoldak) [inline config/database.yml]: move database.yml to gitignore
Add database.yml.example without real credentials

File: config/database.yml [hunk 4]
```diff
@@ -1,25 +1,86 @@
...
+# You can use this database configuration with:
+#
+#   production:
+#     url: <%= ENV['DATABASE_URL'] %>
+#
 production:
   <<: *default
-  database: db/production.sqlite3
+  database: myapp_production
+  username: myapp
```
Comment (AlexeyBoldak) [inline config/database.yml]: This should be ENV variable

File: app/controllers/main_page_controller.rb (hunk 2 above)
Comment (maruraru) [reply app/controllers/main_page_controller.rb]: removed

File: app/controllers/main_page_controller.rb (hunk 1 above)
Comment (maruraru) [reply app/controllers/main_page_controller.rb]: edited

File: config/database.yml (hunk 3 above)
Comment (maruraru) [reply config/database.yml]: We discussed and decided not to remove

File: config/database.yml (hunk 4 above)
Comment (maruraru) [reply config/database.yml]: edited

**Pull request merged**